presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -1037,12 +1039,13 @@ public void close()
         }
     }
 
-    public static Map<String, String> populateTableProperties(ConnectorTableMetadata tableMetadata, FileFormat fileFormat)
+    public static Map<String, String> populateTableProperties(ConnectorTableMetadata tableMetadata, FileFormat fileFormat, ConnectorSession session)
     {
         ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builderWithExpectedSize(5);
         Integer commitRetries = getCommitRetries(tableMetadata.getProperties());
         propertiesBuilder.put(DEFAULT_FILE_FORMAT, fileFormat.toString());
         propertiesBuilder.put(COMMIT_NUM_RETRIES, String.valueOf(commitRetries));
+        propertiesBuilder.put(PARQUET_COMPRESSION, getCompressionCodec(session).getParquetCompressionCodec().get().toString());
```

Comment:
We should add this property only if the file format is Parquet.

I can also see table properties for ORC(`write.orc.compression-codec`) and AVRO(`write.avro.compression-codec`). We should set those as well based on the file format selected.

Reply:
It seems the iceberg community don't have consensus about whether this property should appear when the file format isn't parquet. Currently it would always be set regardless the underlying file formats. Referring to: https://github.com/apache/iceberg/pull/9503#issuecomment-1919889634

And I wonder if this is the ground truth, then should ORC_COMPRESSION and AVRO_COMPRESSION be set as well?

Reply:
From what I understand from the discussion in https://github.com/apache/iceberg/pull/9503, iceberg sets the default value for that property regardless of the file format. But since we are overriding the default value here, should we not update it only if someone is creating/writing the iceberg table in parquet format from Presto? If the format set from Presto is not parquet, we let the property be as-is.

> And I wonder if this is the ground truth, then should ORC_COMPRESSION and AVRO_COMPRESSION be set as well?

I think they are still setting the ORC_COMPRESSION and AVRO_COMPRESSION when the formats are being used: https://github.com/apache/iceberg/blob/f9cdde2516daea6d503798ef62e5869b5e46c31d/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java#L490

Reply:
Yes, you are right, and I agree with you, set the corresponding property base on file format is concise and easy to understand. I just a little confuse that Iceberg itself always set `PARQUET_COMPRESSION` for a new table regardless the underlying file formats, but do not set `ORC_COMPRESSION` and `AVRO_COMPRESSION`.